dev/ci/scripts/utils/ci_utils.sh

```diff
@@ -1,4 +1,13 @@
-#!/bin/env bash
+#!/usr/bin/env bash
+
+# Determine HOMEGFS_ and source machine detection early
+if [[ -z "${HOMEGFS_}" ]]; then
+    SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+    HOMEGFS_="$("${SCRIPT_DIR}/find_homegfs.py")"
```

Comment:
`HOMEgfs_` please (here and elsewhere where their is a need for a local variable to indicate the top of the clone) and does not want to conflict with `HOMEgfs` in `config.base`.

Reply:
Sounds good.